opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -1464,10 +1467,27 @@ public void addEvent(String baseEntityId, JSONObject jsonObject, String syncStat
             }
 
         } catch (Exception e) {
-            Log.e(getClass().getName(), "Exception", e);
+            Timber.e(e);
         }
     }
 
+    public void markEventAsProcessed(String formSubmissionId, String eventID) {
+        try {
+
+            ContentValues values = new ContentValues();
+            values.put(event_column.formSubmissionId.name(), formSubmissionId);
+            values.put(event_column.syncStatus.name(), StringUtils.isNotBlank(eventID) ? BaseRepository.TYPE_Synced : BaseRepository.TYPE_Unsynced);
```

Comment:
Maybe you should update sync Status only if current sync status is  `TYPE_Unsynced`  or `TYPE_unprocessed` and not depend on checking if event has  eventId

Reply:
Updated logic to only update records when the event was in a status of TYPE_unprocessed

Reply:
Resolved